packages/SwingSet/test/test-vpid-kernel.js

```diff
@@ -595,7 +578,7 @@ async function doTest4567(t, which, mode) {
     t.is(clistKernelToVat(kernel, vatA, p1kernel), undefined);
     t.is(clistVatToKernel(kernel, vatA, p1VatA), undefined);
   } else {
-    t.is(inCList(kernel, vatA, p1kernel, p1VatA), true);
+    t.is(inCList(kernel, vatA, p1kernel, p1VatA), false);
```

Comment:
This seems weird.. if the item should be missing from the kernel clist in both `expectRetirement` and `!expectRetirement` cases, then let's move this clause out of the `if` block, rather than making the same test in both arms. But how is it that `inCList()` is expected to return `false` at the same time that `clistKernelToVat`/`clistVatToKernel` are expected to return values? I don't see how that could ever pass, which makes me think that we're no longer exercising the `!expectRetirement` case at all (which is what we want in the long run: we should retire every resolved promise, so sooner or later we *should* remove support for testing the non-retiring case).

Reply:
Right, `expectRetirement` is now unconditionally true, so this second clause isn't being exercised, so the mismatch is not causing a failure.

Let's remove the `expectRetirement` flag entirely, and delete `else` arm.

Reply:
The `expectRetirement` flag is a vestigial leftover from when we were first implementing promise retirement stuff, so that we had the option to run with and without promise retirement (without breaking the tests) until we were satisfied that it was solid.  Now that we've firmly embraced it, I think it can be safely discarded.

Reply:
should we discard it now? or do you have a followup PR to do that?